backend/code/archipelag/market/views.py

```diff
@@ -23,22 +24,41 @@ def get(self, request):
 @login_required
 def market_create(request):
```

Comment:
You can use Django class view: https://docs.djangoproject.com/en/2.0/topics/class-based-views/